crates/pacdef_core/src/backend/actual/fedora.rs

```diff
@@ -0,0 +1,127 @@
+const SWITCHES_MAKE_DEPENDENCY: Switches = &[];
+const SWITCHES_NOCONFIRM: Switches = &["--assumeyes"];
+const SWITCHES_REMOVE: Switches = &["remove"];
+
+const SUPPORTS_AS_DEPENDENCY: bool = true;
+
+impl Backend for Fedora {
+    impl_backend_constants!();
+
+    fn get_all_installed_packages(&self) -> Result<HashSet<Package>> {
+        let mut cmd = Command::new(self.get_binary());
+        cmd.args(self.get_switches_info());
+
+        let output = String::from_utf8(cmd.output()?.stdout)?;
+        let packages: HashSet<Package> = output.lines().map(create_package).collect();
+
+        Ok(packages)
+    }
+
+    fn get_explicitly_installed_packages(&self) -> Result<HashSet<Package>> {
+        let mut cmd = Command::new(self.get_binary());
+        cmd.args([
+            "repoquery",
+            "--userinstalled",
+            "--queryformat",
+            "%{from_repo}/%{name}",
+        ]);
+
+        let output = String::from_utf8(cmd.output()?.stdout)?;
+        let packages: HashSet<Package> = output.lines().map(create_package).collect();
+
+        Ok(packages)
+    }
+
+    /// Install the specified packages.
+    fn install_packages(&self, packages: &[Package], noconfirm: bool) -> Result<()> {
+        let mut cmd = Command::new("sudo");
+        cmd.arg(self.get_binary());
+        cmd.args(self.get_switches_install());
+
+        if noconfirm {
+            cmd.args(self.get_switches_noconfirm());
+        }
+
+        for p in packages {
+            cmd.arg(&p.name);
+            if let Some(repo) = p.repo.as_ref() {
+                cmd.args(["--repo", repo]);
+            }
+        }
+
+        // add these two repositories as these are needed for many dependencies
+        cmd.args(["--repo", "updates"]);
+        cmd.args(["--repo", "fedora"]);
+
+        run_external_command(cmd)
+    }
+
+    fn remove_packages(&self, packages: &[Package], noconfirm: bool) -> Result<()> {
+        let mut cmd = Command::new("sudo");
+        cmd.arg(self.get_binary());
+        cmd.args(self.get_switches_remove());
+
+        if noconfirm {
+            cmd.args(self.get_switches_noconfirm());
+        }
+
+        for p in packages {
+            cmd.arg(&p.name);
+        }
+
+        run_external_command(cmd)
+    }
+
+    fn make_dependency(&self, _: &[Package]) -> Result<()> {
+        panic!("Not supported by the package manager!")
+    }
+}
+
+impl Fedora {
+    pub fn new() -> Self {
+        Self {
+            packages: HashSet::new(),
+        }
+    }
+}
+
+fn create_package(package: &str) -> Package {
+    if (package.contains("koji")
+        || package.contains("fedora")
+        || package.contains("updates")
+        || package.contains("anaconda")
+        || package.contains('@'))
+        && !package.contains("copr")
+    {
+        let package = package.split('/').nth(1).expect("Cannot be empty!");
+        package.into()
+    } else {
+        package.into()
+    }
+}
```

Comment:
You can write that this way a bit more elegantly.

```rust
fn create_package(package: &str) -> Package {
    if SPECIAL_PACKAGES
        .iter()
        .any(|special| package.contains(*special))
    {
        let package = package.split('/').nth(1).expect("Cannot be empty!");
        package.into()
    } else {
        package.into()
    }
}
```

and then add the following constant at the top of the file.

```rust
/// put a good docstring here
const SPECIAL_PACKAGES: [&str; 6] = ["koji", "fedora", "updates", "anaconda", "@", "copr"];
```

and maybe think of a better name than SPECIAL_PACKAGE :)